src/modules/esl-media-query/core/esl-media-rule-list.ts

```diff
@@ -1,83 +1,131 @@
-  private readonly _rules: ESLMediaRule<T>[];
-
-  public static STRING_PARSER = (val: string) => val;
+@ExportNs('MediaRuleList')
+export class ESLMediaRuleList<T = any> extends Observable<RuleChangedCallback<T>> {
+  /**
+   * String value parser (used as a default)
+   * @returns value string as it is
+   */
+  public static STRING_PARSER: RulePayloadParser<string> = (val: string): string => val;
+  /**
+   * Object value parser. Uses {@link evaluate} to parse value
+   * @returns value - parsed JS Object
+   */
   public static OBJECT_PARSER = <U>(val: string): U | undefined => evaluate(val);
 
-  private static parseRules<U>(str: string, parser: PayloadParser<U>): ESLMediaRule<U>[] {
-    const parts = str.split('|');
-    const rules: ESLMediaRule<U>[] = [];
-    parts.forEach((_lex: string) => {
-      const lex = _lex.trim();
-      if (!lex) {
-        return;
-      }
-      if (lex.indexOf('=>') === -1) {
-        const value = parser(lex);
-        // Default rule should have lower priority
-        typeof value !== 'undefined' && rules.unshift(ESLMediaRule.all(value));
-      } else {
-        const rule = ESLMediaRule.parse(lex, parser);
-        rule && rules.push(rule);
-      }
+  /**
+   * Creates `ESLMediaRuleList` from string query representation
+   * Uses exact strings as rule list values
+   * @param query - query string
+   */
+  public static parse(query: string): ESLMediaRuleList<string>;
+  /**
+   * Creates `ESLMediaRuleList` from string query representation
+   * @param query - query string
+   * @param parser - value parser function. See built in {@link ESLMediaRuleList.OBJECT_PARSER} or {@link ESLMediaRuleList.STRING_PARSER} (default)
+   */
+  public static parse<U>(query: string, parser: RulePayloadParser<U>): ESLMediaRuleList<U>;
+  public static parse(query: string, parser: RulePayloadParser<any> = ESLMediaRuleList.STRING_PARSER): ESLMediaRuleList<any> {
+    const rules: ESLMediaRule<any>[] = [];
+    query.split('|').forEach((part: string) => {
+      const lex = part.trim();
+      const rule = ESLMediaRule.parse(lex, parser);
+      if (!rule) return;
+      rule.default ? rules.unshift(rule) : rules.push(rule);
     });
-    return rules;
+    return new ESLMediaRuleList(rules);
   }
 
-  public static parse<U>(query: string, parser: PayloadParser<U>) {
-    return new ESLMediaRuleList<U>(query, parser);
+  /**
+   * Creates `ESLMediaRuleList` from two strings with a value  and conditions tuple
+   *
+   * @example
+   * ESLMediaRuleList.fromTuple('1|2|3|4|5', '@XS|@SM|@MD|@LG|@XL')
+   *
+   * @param values - values tuple string (uses '|' as separator)
+   * @param mask - media conditions tuple string (uses '|' as separator)
+   */
+  public static parseTuple(values: string, mask: string) {
+    const valueList = values.split('|');
+    const conditions = mask.split('|');
+    if (valueList.length !== conditions.length) throw new Error('Value doesn\'t correspond to mask');
```

Comment:
'Value doesn't correspond to mask'

Reply:
@yadamskaya could you please clarify the point? Did you mean to replace `value` with `values`?
PS: I'll fix this point in "followups"